### DISCUSSION TITLE
Error Handling Issue in Frontend Form

### DISCUSSION THREAD

**Discussion by outline4** [solspace/craft-calendar#262] [Questions & Answers]
**Describe the bug or issue you're experiencing**

**Steps to reproduce**

1. I've set up my frontend form with the aid of the "edit event" template. All works fine. 
2. As soon as an error is displayed the rest the data that wasn't already saved is deleted. 

This happens with regular fields. 

For instance: my field "event_meeting_point" has a limit of 128 Characters. 

`<input type="text" name="fields[event_meeting_point]" id="event_meeting_point" value="{{ event.event_meeting_point }}">`

If I enter more characters than allowed by the field, the following error is correctly displayed: 

"Meeting point has a maximum of 128 Characters"

I find myself with empty fields since the last sucessful save. 

**Expected behavior**

It stores the values without errors and only then display the error. It should even preserve the value of the erronous field. 

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 4.5.13
- Calendar Version: 4.0.11
- Calendar Edition: Pro
- Fresh Install or Upgrade: Fresh

**Comment by outline4** [comment]
I can confirm the same behaviour! 

To prevent this, we've split up the form into multiple steps, so only the data of the current step isn't saved in case of an error. It's not perfect, but at least we don't lose that much data. 

Hope this gets fixed soon!

**Comment by kjmartens** [comment]
Sorry for the delay @outline4,

If I understand correctly, I think the issue is that you would need the form to submit via AJAX instead? This would preserve the values of the other fields without losing them. 🙂
